### PR TITLE
create_team_admin

### PR DESCRIPTION
**Pull request by rofiuddin**

### CODE REVIEW

File: app/controllers/admin/teams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      redirect_to admin_team_path @team
+    else
+      render :new
+    end
+  end
+
+  private
+  def team_params
+    params.require(:team).permit :name, :description,:leader_id,
+                                 :team_id, user_ids: [],
```
Comment (MuhammadTamzid) [inline app/controllers/admin/teams_controller.rb]: indent

File: app/controllers/admin/teams_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                                 team_users_attributes: [:id, :user_id, :team_id, :_destroy]
```
Comment (MuhammadTamzid) [inline app/controllers/admin/teams_controller.rb]: indent

File: app/models/team.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class Team < ActiveRecord::Base
-  has_many :users
-  has_many :projects, dependent: :destroy
+  has_many :team_users, dependent: :destroy
+  has_many :users, through: :team_users
+
+  belongs_to :leader, class_name: "User"
```
Comment (MuhammadTamzid) [inline app/models/team.rb]: "User" => User.name

File: app/models/team_user.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
 class TeamUser < ActiveRecord::Base
   belongs_to :user
   belongs_to :team
+
+  validates :team_id, presence: true
```
Comment (MuhammadTamzid) [inline app/models/team_user.rb]: validates :team, presence: true

File: app/models/team_user.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
...
+  validates :user_id, presence: true
```
Comment (MuhammadTamzid) [inline app/models/team_user.rb]: validates :user, presence: true

File: app/models/user.rb [hunk 6]
```diff
@@ -11,4 +11,5 @@ class User < ActiveRecord::Base
   belongs_to :team
 
   accepts_nested_attributes_for :position_users, allow_destroy: true
+  scope :leader_not_in_team, ->{where Settings.leader.not_in_team}
```
Comment (MuhammadTamzid) [inline app/models/user.rb]: it is better if add a new line before this block

File: app/controllers/admin/teams_controller.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  def new
+    @team = Team.new
+    @leaders = User.leader_not_in_team
+    @team.team_users.build
+  end
+
+  def create
+    @team = Team.new team_params
+    if @team.save
+      @team.users << @team.leader
```
Comment (dieunb) [inline app/controllers/admin/teams_controller.rb]: why need it?

File: config/settings.yml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+leader:
```
Comment (dieunb) [inline config/settings.yml]: no need, should move it into model

File: app/controllers/admin/teams_controller.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    else
+      render :new
+    end
+  end
+
+  private
+  def team_params
+    params.require(:team).permit :name, :description,:leader_id,
+      :team_id, user_ids: [],
+        team_users_attributes: [:id, :user_id, :team_id, :_destroy]
```
Comment (MuhammadTamzid) [inline app/controllers/admin/teams_controller.rb]: indent

Comment (rofiuddin) [inline app/controllers/admin/teams_controller.rb]: bhai two space toh, one to [code]:[/code] and other to [code]t[/code] :/

File: config/settings.yml (hunk 8 above)
Comment (rofiuddin) [inline config/settings.yml]: okay.

File: app/controllers/admin/teams_controller.rb (hunk 7 above)
Comment (rofiuddin) [inline app/controllers/admin/teams_controller.rb]: I want to save leader into team_user also. should remove?

File: app/controllers/admin/teams_controller.rb (hunk 9 above)
Comment (MuhammadTamzid) [inline app/controllers/admin/teams_controller.rb]: :team_id, user_ids: [],
team_users_attributes: [:id, :user_id, :team_id, :_destroy]

File: app/controllers/admin/teams_controller.rb (hunk 7 above)
Comment (dieunb) [inline app/controllers/admin/teams_controller.rb]: i saw in your team model
You have:
```
has_many :users
belongs_to :leader
```

So no need push it into users

File: app/models/team.rb [hunk 10]
```diff
@@ -1,4 +1,8 @@
 class Team < ActiveRecord::Base
-  has_many :users
-  has_many :projects, dependent: :destroy
+  has_many :team_users, dependent: :destroy
+  has_many :users, through: :team_users
+
+  belongs_to :leader, foreign_key: "user_id", class_name: "User"
```
Comment (dieunb) [inline app/models/team.rb]: `belongs_to :leader, foreign_key: :user_id, class_name: User.name`

**Pull request merged**